### PR TITLE
send supplemental context strategy telemetry field

### PR DESCRIPTION
**Pull request by Will-ShaoHua**
We are already able to analyze acceptance rate on either CrossFile or UTG or an aggregated number. Now we need more granular metrics for UTG `resolve source file by file name` vs. `resolve source file by file content` to help us understanding deeper the effectiveness of UTG, thus adding a new field `codewhispererSupplementalContextStrategyId` field and there are 4 values for now
`Empty`, `ByName`, `ByContent`, `OpenTabs_BM25`

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)

## Checklist
- [ ] My code follows the code style of this project
- [ ] I have added tests to cover my changes
- [ ] A short description of the change has been added to the **[CHANGELOG](https://github.com/aws/aws-toolkit-jetbrains/blob/master/CONTRIBUTING.md#contributing-via-pull-requests)** if the change is customer-facing in the IDE.
- [ ] I have added metrics for my changes (if required)
 
## License
I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: jetbrains-core/src/software/aws/toolkits/jetbrains/services/codewhisperer/util/CodeWhispererFileContextProvider.kt [hunk 1]
```diff
@@ -236,20 +233,31 @@ class DefaultCodeWhispererFileContextProvider(private val project: Project) : Fi
...
+        val focalFile = utgCandidateResult.vfile
+        val strategy = utgCandidateResult.strategy
 
         return focalFile?.let { file ->
             runReadAction {
                 val relativePath = contentRootPathProvider.getPathToElement(project, file, null) ?: file.path
                 val content = file.content()
 
-                if (content.isBlank()) {
+                val utgContext = if (content.isBlank()) {
```
Comment (andrewyuq) [inline jetbrains-core/src/software/aws/toolkits/jetbrains/services/codewhisperer/util/CodeWhispererFileContextProvider.kt:260]: `utgContents`?

**Review by andrewyuq — APPROVED** [review]

File: jetbrains-core/src/software/aws/toolkits/jetbrains/services/codewhisperer/util/CodeWhispererFileContextProvider.kt (hunk 1 above)
Comment (Will-ShaoHua) [reply jetbrains-core/src/software/aws/toolkits/jetbrains/services/codewhisperer/util/CodeWhispererFileContextProvider.kt:260]: ahh either works

Comment (Will-ShaoHua) [reply jetbrains-core/src/software/aws/toolkits/jetbrains/services/codewhisperer/util/CodeWhispererFileContextProvider.kt:260]: I was thinking utgContext

Comment (Will-ShaoHua) [reply jetbrains-core/src/software/aws/toolkits/jetbrains/services/codewhisperer/util/CodeWhispererFileContextProvider.kt:260]: as opposed to crossFileContext

**Review by rli — APPROVED** [review]

**Pull request merged**